### PR TITLE
REPO: Perform initial setup of the software-common repository

### PR DESCRIPTION
**Pull request by sentree**
**This pull request is part of a series of four. All four are on different repositories and it is crucial that they are merged at the same time to keep the install process working!**

These are the other pull requests to look at:
* [installer #2](https://github.com/uf-mil/installer/pull/2)
* [Sub8 #199](https://github.com/uf-mil/Sub8/pull/199)
* [Navigator #192](https://github.com/uf-mil/Navigator/pull/192)

There are a few things that were done on most if not all of these repositories:
* Some outdated code and dependencies were removed
* Aliases were moved to a bash_aliases.sh file in the scripts directory
    * This is sourced by the ~/.milrc file now created by the install script
    * General aliases are here, specific ones stay with their respective projects
* All roads (or readme files, that is) now lead to the software common wiki (which I will be working on in the next few days)
    * Check out the vehicle images!

Now on to software-common specifics:
* The bag backup script was moved over and should run automatically once the sub OS is fully set up (definitely before Saturday)
* Common repository dependencies are now submodules instead of being pulled separately
    * Allows us to easily version control them (i.e. if we roll one back, that can be committed)
    * I recommend purging your workspace and running the install script after this goes through or there will be duplicate packages
* Two line bash is now a version controlled script, it will be installed next time the install script is run (I included a way to disable it, assuming I'm not the only one that doesn't like it [am I??])
* The ros_connect script is now an absolute beast
    * A hostname or IP can be passed with the -n argument to skip the ~2 seconds of host discovery
    * Persistence across shells is supported (run once and the master that is selected carries over to new shells)
    * Persistence can be toggled on the fly
    * One-time runs are supported with the -o argument (i.e. if I want just the current shell to have example.com as a ros master, not have it persist to new ones)
    * Yes, I left rsub and rnav as aliases for those stuck in their ways (ahem, @mattlangford)

### CODE REVIEW

**Comment by sentree** [comment]
@DSsoto: IT'S HAPPENING **osprey*

**Comment by sentree** [comment]
Once these are merged, I recommend that we rename a few of the repositories for the sake of standardization. These are just suggestions of course, but I thought I'd bring up the point:
* software-common -> mil_common
* rawgps-tools -> rawgps_tools
* Sub8 -> SubjuGator or subjugator
* Navigator -> NaviGator or navigator

File: scripts/two_line_bash.sh
Comment (mattlangford) [inline scripts/two_line_bash.sh:24]: Ahem

File: scripts/bash_aliases.sh
Comment (mattlangford) [inline scripts/bash_aliases.sh:20]: Can you add a -j8 in there too?

**Review by mattlangford — COMMENTED** [review]
I like it! We're in the future.

I'll let a more bashy person approve this (@ironmig)

File: scripts/two_line_bash.sh
Comment (sentree) [reply scripts/two_line_bash.sh:24]: Relax, it's not the default!

File: scripts/bash_aliases.sh
Comment (sentree) [reply scripts/bash_aliases.sh:20]: I was under the impression that catkin_make automatically detected the number of available threads. Is this not the case?

Comment (DSsoto) [reply scripts/bash_aliases.sh:20]: Not the case.

**Review by DSsoto — APPROVED** [review]
Looks good, I have no qualms with any of this.

**Comment by DSsoto** [comment]
I agree with @whispercoros  on the renaming of the repos, just to have standardized naming. I would also prefer SubjuGator to subjugator since it's a proper name. @mattlangford, thoughts?

**Pull request merged**
